nix/modules/lanzaboote.nix

```diff
@@ -74,5 +74,20 @@ in
           /nix/var/nix/profiles/system-*-link
       '';
     };
+
+    systemd.services.fwupd = lib.mkIf config.services.fwupd.enable {
+      # Tell fwupd to load its efi files from /run
+      environment.FWUPD_EFIAPPDIR = "/run/fwupd-efi";
+      serviceConfig.RuntimeDirectory = "fwupd-efi";
```

Comment:
@dasJ, just curious is `serviceConfig.RuntimeDirectory` needed?

This seems to clash because the built-in unit already sets `RuntimeDirectory` and so systemd appends them both separated by a `:`, which fwupd doesn't understand how to process multiple runtime dirs and so assumes `/run/fwupd-efi:/run/motd.d` is literal and makes the motd file in that suspicious-looking directory `/run/fwupd-efi:/run/motd.d/85-fwupd`

Reply:
Ah I think it was to make sure that directory exists. Since it makes more problems than it solves, I've removed it and added a `mkdir -p /run/fwupd-efi` to the pre-start script

Let me know if my analysis there is wrong @dasJ or if this is appropriate

Reply:
Yeah that works as well but I hate the approach of doing mkdir/chmod/chown in prestart. That's why I did it this way. We could also set `PrivateTmp` and just write to /tmp instead?

Reply:
I like `/run` better than `/tmp` (since it's supposed to be available for the entire lifetime of the daemon and isn't really temporary), but I'll try to think of options that don't use mkdir if you want

Reply:
Here's a diff for that latter option, for comparison:

```diff
diff --git a/nix/modules/lanzaboote.nix b/nix/modules/lanzaboote.nix
index 02f8f3f..0514548 100644
--- a/nix/modules/lanzaboote.nix
+++ b/nix/modules/lanzaboote.nix
@@ -78,9 +78,16 @@ in
     systemd.services.fwupd = lib.mkIf config.services.fwupd.enable {
       # Tell fwupd to load its efi files from /run
       environment.FWUPD_EFIAPPDIR = "/run/fwupd-efi";
+    };
+
+    systemd.services.fwupd-efi = lib.mkIf config.services.fwupd.enable {
+      # Exist with the lifetime of the fwupd service
+      boundBy = [ "fwupd.service" ];
+      before = [ "fwupd.service" ];
+      # Create runtime directory for signed EFI app
+      serviceConfig.RuntimeDirectory = "fwupd-efi";
       # Place the fwupd efi files in /run and sign them
-      preStart = ''
-        mkdir -p /run/fwupd-efi
+      script = ''
         cp ${config.services.fwupd.package.fwupd-efi}/libexec/fwupd/efi/fwupd*.efi /run/fwupd-efi/
         ${pkgs.sbsigntool}/bin/sbsign --key '${cfg.privateKeyFile}' --cert '${cfg.publicKeyFile}' /run/fwupd-efi/fwupd*.efi
       '';
```